Lib/asyncio/__main__.py

```diff
@@ -75,12 +77,36 @@ def run(self):
                 f'Use "await" directly instead of "asyncio.run()".\n'
                 f'Type "help", "copyright", "credits" or "license" '
                 f'for more information.\n'
-                f'{getattr(sys, "ps1", ">>> ")}import asyncio'
             )
+            exit_message = 'exiting asyncio REPL...'
+
+            console.write(banner)
+            if startup_path := os.getenv("PYTHONSTARTUP"):
+                import tokenize
+                with tokenize.open(startup_path) as f:
+                    startup_code = compile(f.read(), startup_path, "exec")
+                    exec(startup_code, console.locals)
 
-            console.interact(
-                banner=banner,
-                exitmsg='exiting asyncio REPL...')
+            try:
+                import errno
+                if not os.isatty(sys.stdin.fileno()):
+                    raise OSError(errno.ENOTTY, "tty required", "stdin")
+
+                # This import will fail on operating systems with no termios.
+                from _pyrepl.simple_interact import (
+                    check,
+                    run_multiline_interactive_console,
+                )
+                if err := check():
+                    raise RuntimeError(err)
+            except Exception as e:
+                console.interact(banner="", exitmsg=exit_message)
+            else:
+                try:
+                    run_multiline_interactive_console(console=console)
+                except BaseException:
+                    console.showtraceback()
```

Comment:
It looks like this prints a traceback when the user types `quit` or `exit`, unlike the default (non-asyncio) colored REPL. The asyncio REPL exits quietly when you type `quit()` or `exit()`.

Reply:
The new REPL also exits quietly when you type in `exit()` with parentheses. The difference is that the non-parenthesized code is executing on the wrong thread. This needs to be fixed.

Reply:
I changed my mind. Most commands like `help` and `clear` actually should be executed on the interactive thread. Instead, I made SystemExit support better across the board.